src/editor/classes/MessageHandler.js

```diff
@@ -0,0 +1,457 @@
+    {
+        var cur = editor.getCursor();
+        var line = editor.getLine(cur.line);
+
+        var strL = line.substring(0, cur.ch);
+        var strR = line.substring(cur.ch);
+        var regexL = /[^\w]?(\w*)$/;
+        var regexR = /^(\w*)[^\w]?/;
+        var matchL = regexL.exec(strL);
+        var matchR = regexR.exec(strR);
+
+        var word = "";
+        if (matchL !== null) word += matchL[1];
+        if (matchR !== null) word += matchR[1];
+
+        return word;
+    }
+
+    C_CMD_DUPLICATE_LINE(data) 
+    {
+        var cur = editor.getCursor();
+        var line = editor.getLine(cur.line);
+        var pos = {line: cur.line, ch: line.length};
+        editor.replaceRange('\n' + line, pos);
+    }
+
+    C_CMD_MOVE_LINE_UP(data) 
+    {
+        var cur = editor.getCursor();
+    
+        // check previous line is not beginning of the document
+        if ((cur.line - 1) < 0) {
+            return;
+        }
+    
+        var line = editor.getLine(cur.line) + '\n' + editor.getLine(cur.line-1);
+        var from = { line: cur.line - 1, ch: 0           };
+        var to   = { line: cur.line,     ch: line.length };
+    
+        editor.replaceRange(line, from, to);
+        editor.setCursor(cur.line - 1, cur.ch );
+    }
+
+    C_CMD_MOVE_LINE_DOWN(data) 
+    {
+        var cur = editor.getCursor();
+    
+        // check that next line is not past end of document
+        if ((cur.line + 1) == editor.lineCount() )  {
+            return;
+        }
+    
+        var line = editor.getLine(cur.line + 1) + '\n' + editor.getLine(cur.line);
+        var from = { line: cur.line,     ch: 0           };
+        var to   = { line: cur.line + 1, ch: line.length };
+    
+        editor.replaceRange(line, from, to);
+        editor.setCursor(cur.line + 1, cur.ch );
+    }
+
+
+    C_CMD_DELETE_LINE(data) 
+    {
+        editor.execCommand("deleteLine");
+        editor.changeGeneration(true);
+    }
+
+    C_CMD_TRIM_LEADING_TRAILING_SPACE(data) 
+    {
+        editLines(function (x) { return x.trim(); });
+    }
+
+    C_CMD_TRIM_TRAILING_SPACE(data) 
+    {
+        editLines(function (x) { return x.replace(/\s+$/, ""); });
+    }
+
+    C_CMD_TRIM_LEADING_SPACE(data) 
+    {
+        editLines(function (x) { return x.replace(/^\s+/, ""); });
+    }
+
+    C_CMD_TAB_TO_SPACE(data) 
+    {
+        editLines(function (x) {
+            tabToSpaceCounter = 0
+            var tabSz = editor.getOption("tabSize")
+
+            return x.replace(/\t/g, function(match, offset) {
+                return tabToSpaceHelper(match, offset, tabSz)
+            });
+
+        });
+    }
+
+    C_CMD_SPACE_TO_TAB_ALL(data) 
+    {
+        editLines(function (x) {
+            App.helpers.spaceToTabCounter = 0
+            var tabSz = editor.getOption("tabSize")
+
+            return x.replace(/ +/g, function(match, offset) {
+                return App.helpers.spaceToTab(match, offset, tabSz)
+            });
```

Comment:
My main point was that tightly coupled code should either be near each other or should be very clear to reduce code spaghetti. So if you think that solves it, give it a try.